### ISSUE TITLE
[BUG] Auto-backport is failing: permission denied

### ISSUE THREAD

**Issue by dblock** [opensearch-project/OpenSearch#1712]
**Describe the bug**

```
remote: Permission to opensearch-project/OpenSearch.git denied to github-actions[bot].
```

**To Reproduce**
1. Label an issue backport 1.x.
2. See failed job.

https://github.com/opensearch-project/OpenSearch/runs/4509880595?check_suite_focus=true

**Comment by dblock** [comment]
@VachaShah Do you know how to fix this?

**Comment by VachaShah** [comment]
Tried the approach where in the workflow, we fork the repo and then create a branch to backport, cherry-pick the commit and create a PR from it. But the Github actions was not able to fork a repo, got error `failed to fork: HTTP 403: Resource not accessible by integration` while doing so.

**Comment by setiah** [comment]
@VachaShah try with a _pre-created fork_ (say backport/OpenSearch) to which this backport GH has access, and use the `checkout@v2` action to backport, push and raise PR from this backport fork. This would also help keep the permissions of this GH restricted to the backport fork alone and track the workflow retries better.

**Comment by VachaShah** [comment]
@setiah Wouldn't the pre-created fork have to belong to a user?

**Comment by VachaShah** [comment]
Update, the following things are needed to fix the permissions issue:

1. The token needs write permissions for pull requests and content in order to create pull requests on the repos. This is fixed in PR #1845.
2. The GHA for backport creates branches (`backport-<PR-number>-to-<release-branch>`) to cherry-pick commits for backport. Hence, the branch permissions need to be defined for such branches. In order to better manage permissions, I have PRed the change to allow custom branch names in the GHA repo and use it in the OpenSearch repo. Details: #1862.

**Comment by VachaShah** [comment]
This issue is resolved and the backport workflow is able to create PRs. Example PR: #1959
